src/util/carddb.ts

```diff
@@ -217,37 +246,45 @@ export function getRelatedCards(oracleId: string): Record<string, Record<string,
     };
   }
 
+  const mapper = (oracleIndex: number) => {
+    return indexToReasonableWithPrintingPreference(oracleIndex, printingPreference);
+  };
+
   return {
     cubedWith: {
-      top: related.cubedWith.top.map(indexToReasonable),
-      creatures: related.cubedWith.creatures.map(indexToReasonable),
-      spells: related.cubedWith.spells.map(indexToReasonable),
-      other: related.cubedWith.other.map(indexToReasonable),
+      top: related.cubedWith.top.map(mapper),
+      creatures: related.cubedWith.creatures.map(mapper),
+      spells: related.cubedWith.spells.map(mapper),
+      other: related.cubedWith.other.map(mapper),
     },
     draftedWith: {
-      top: related.draftedWith.top.map(indexToReasonable),
-      creatures: related.draftedWith.creatures.map(indexToReasonable),
-      spells: related.draftedWith.spells.map(indexToReasonable),
-      other: related.draftedWith.other.map(indexToReasonable),
+      top: related.draftedWith.top.map(mapper),
+      creatures: related.draftedWith.creatures.map(mapper),
+      spells: related.draftedWith.spells.map(mapper),
+      other: related.draftedWith.other.map(mapper),
     },
     synergistic: {
-      top: related.synergistic.top.map(indexToReasonable),
-      creatures: related.synergistic.creatures.map(indexToReasonable),
-      spells: related.synergistic.spells.map(indexToReasonable),
-      other: related.synergistic.other.map(indexToReasonable),
+      top: related.synergistic.top.map(mapper),
+      creatures: related.synergistic.creatures.map(mapper),
+      spells: related.synergistic.spells.map(mapper),
+      other: related.synergistic.other.map(mapper),
     },
   };
 }
 
 // if the oracle id is not in the training data, we will use the similar card in the metadata dict
-export function getOracleForMl(oracleId: string): string {
+export function getOracleForMl(oracleId: string, printingPreference: PrintingPreference | null): string {
```

Comment:
getOracleForMl is called by drafting functions and there they don't need printing preference. After determining the oracle id/rating, usually there will be sorting or filtering against a card set which are already specific printings